src/bank02.asm

```diff
@@ -8130,29 +8126,194 @@ titleScreenLicenseText:
+    push HL
+    ld   HL, wHPMPTileBuffer
+    ld   D, B
+    add  HL, DE
+    ld   D, H
+    ld   E, L
+    pop  HL
+    push BC
+    push DE
+
+    ; Load up the digits on the stack
+.divmod_and_store_number:
+    ld   A, $0a
+    push BC
+    push DE
+    call divMod
+    pop  DE
+    pop  BC
+    push AF
+    inc  B
+    ld   A, H
+    or   A, L
+    jr   NZ, .divmod_and_store_number
+
+    ; Reduce second nibble of C by num digits found
+    ld   A, C
+    sub  A, B
+    ld   C, A
+
+    ; Increment DE if in swap mode
+    ld   H, D
+    ld   L, E
+    ld   D, $00
+    and  A, $f0
+    ld   E, A
+    add  HL, DE
+    ld   D, H
+    ld   E, L
+
+    ; Stack holds B digits to pop, transfer them
+ .grab_digit_and_transfer:
+    ld   HL, $6800
+    pop  AF ; grab digit
+    swap A
+    push DE
+    ld   D, $00
+    ld   E, A
+    add  HL, DE
+    pop  DE
+    ld   A, $08
+    push BC
+    ld   B, $10
+    call copyShiftBankAfromHLtoDE
+    pop  BC
+    dec  B
+    jr   NZ, .grab_digit_and_transfer
+
+    ; Clear remaining tiles
+    ld   A, C
+    and  A, $0f
+    jr   Z, .check_swap
+
+    ld   H, D
+    ld   L, E
+    ld   B, A
+    xor  A, A
+ .clear_remaining_tiles:
+    call memsetTileWithA
+    dec  B
+    jr   NZ, .clear_remaining_tiles
+
+    ; Shifted tiles are in WRAM, check if swap needed
+ .check_swap:
+    pop  DE
+    ld   A, C
+    ld   L, C
+    pop  BC
+    and  A, $f0
+    ret  Z
+
+    ; We need to swap back tiles, prepare...
+    ld   A, C
+    sub  A, L
+    ld   B, A ; B holds tiles to swap
+
+    jp   swapBackHalfTiles
+
+; Swaps tiles back across half tile boundaries
+; B holds number of tiles to shift
+; DE holds address of first tile
+swapBackHalfTiles:
+    swap B ; now B holds bytes to swap
+
+    ; Clear the first tile (it was left untouched so far)
+    ld   H, D
+    ld   L, E
+    xor  A, A
+    call memsetTileWithA
+
+.shift_4px:
+    ld   A, [HL]
+    swap A
+    and  A, $0f
+    ld   C, A
+    ld   A, [DE]
+    or   A, C
+    ld   [DE], A
+    ld   A, [HL]
+    swap A
+    and  A, $f0
+    ld   [HL+], A
+    inc  DE
+    dec  B
+    jr   NZ, .shift_4px
+    ret
+
+; Requests transfer of status bar WRAM tiles to VRAM
+; D holds VRAM tile position
+; E holds the WRAM offset in the first nibble (16-byte aligned)
+; E holds the number of tiles to transfer in the second nibble
+requestVRAMStatusBarTransfer:
+    ; Short circuit if nothing to do, else load number of tiles in B
+    ld   A, E
+    and  A, $0f
+    ret  Z
+    ld   B, A
+
+    ; Prep source and destination addresses
+    ld   HL, wHPMPTileBuffer
+    ld   C, D
+    ld   D, $00
+    ld   A, E
+    and  A, $f0
+    ld   E, A
+    add  HL, DE
+    ld   D, $90
+    ld   E, C
+    xor  A, A
+.request_loop:
+    push BC
+    push DE
+    push HL
+    call addTileGraphicCopyRequest
+    pop  HL
+    ld   DE, $0010
+    add  HL, DE
+    ld   B, H
+    ld   C, L
+    pop  HL
+    add  HL, DE
+    ld   D, H
+    ld   E, L
+    ld   H, B
+    ld   L, C
```

Comment:
Spent a bit longer on this than I should have, but I think it can be improved by loading the constant into `bc`, and pushing `hl` before `de`. Something like:

`    push BC`
`    push HL`
`    push DE`
`    call addTileGraphicCopyRequest`
`    ld BC, $0010`
`    pop HL`
`    add HL, BC`
`    ld D, H`
`    ld E, L`
`    pop HL`
`    add HL, BC`

Reply:
Good call, fixed!